Test scalar trial stress update

dschwen opened the pull request:
Run this through testing....

dschwen committed:
Scalar trial stress update (#173)

dschwen commented:
The new test now runs with
```
  Result has extra variable input.state/se.
  Result has extra variable output.forces/s.
  Result has wrong value for variable output.state/ep_rate. Maximum mixed
  difference = 1.255339e-01 > atol = 1.000000e-08
  Result is missing variable output.state/s.
```

I had to change `state/s` to `forces/s` to make it work, but that that breaks the expected outputs.  I wonder why the `state/ep_rate` result is so wrong?

hugary1995 commented:
As long as `state/S` (the stress tensor) matches the other version, I'd call this good.

hugary1995 committed:
Modify the input file slightly so that the gold file can be reused

reverendbedford commented on include/neml2/models/ScalarTrialStressUpdate.h:
Name should probably indicate the restrictions of when this can be used (isotropic elasticity, J2 flow function).  I also mentioned to Gary that I'd like some way to enforce those restrictions, i.e. throw an error if you're not using isotropic elasticity and the J2 yield surface.  Maybe he has some idea how to do that.

hugary1995 replied:
Name is descriptive enough now.

> Maybe he has some idea how to do that.

I think "action" will take care of that, eventually...

reverendbedford replied:
Ya, okay the name is enough for now.

reverendbedford commented on src/neml2/models/ScalarTrialStressUpdate.cxx:
```diff
@@ -0,0 +1,55 @@
+}
+
+void
+ScalarTrialStressUpdate::set_value(bool out, bool /*dout_din*/, bool /*d2out_din2*/)
```
Don't we need the chain rule derivatives here?

dschwen replied:
Yeah, we do. I though I had added use ad options here...

dschwen replied:
And I did. What are you saying is missing then?

reverendbedford replied:
Before we merge it would be nice to have the analytic derivatives included, like all the other models.

dschwen replied:
I implemented the derivatives

dschwen commented:
The code coverage test seems broken ("Bad credentials")

reverendbedford commented:
Alright, let's figure out why the stupid action won't run.

hugary1995 commented:
We ran into this issue with Lynn's PR before. I think we ended up making him a collaborator so the tests could run. But maybe we should figure out how to get the tests to run for external PRs.

hugary1995 commented:
@dschwen we shall add unit tests for the new models. I'll take care of those.

reverendbedford commented:
> We ran into this issue with Lynn's PR before. I think we ended up making him a collaborator so the tests could run. But maybe we should figure out how to get the tests to run for external PRs.

This is indeed the problem, I think it fails when you're submitting as a PR from a fork, rather than directly on this repo.  I added @dschwen as a collaborator on this project but this is still setup as a PR from his fork.

Let's see if I can fix it...

hugary1995 committed:
A few changes after I review the PR:

- Renamed J2FlowDirection to AssociativeJ2FlowDirection to emphasize the assumption
- Renamed ScalarTrialStressUpdate to LinearIsotropicElasticJ2TrialStressUpdate to emphasize the assumptions
- Misc changes in the source files including documentation, doxygen docstrings, missing derivatives, etc.
- Add unit tests for the two new objects

hugary1995 commented:
We need to set `deploy-repository` and `token` for the documentation preview to be deployed. See https://github.com/marketplace/actions/deploy-pr-preview

reverendbedford commented:
@dschwen could you please move this PR over to our new location here: https://github.com/applied-material-modeling/neml2  @hugary1995 claims the tests will also run fine now from your fork.

hugary1995 commented:
> @dschwen could you please move this PR over to our new location here: https://github.com/applied-material-modeling/neml2 @hugary1995 claims the tests will also run fine now from your fork.

Not yet... hold on to this PR

reverendbedford commented:
@dschwen in case you couldn't tell we just moved the whole repository.  So it just moved it automatically.

hugary1995 committed:
clang-tidy
